src/DominantJuice.sol

```diff
@@ -0,0 +1,399 @@
+    }
+
+    /**
+     * @notice Received hook from the payment terminal after a redemption. If cycle meets project target,
+     * then this function cannot be called, since pauseRedemptions will be true. If the cycle fails to meet
+     * its target, then this function will be callable for pledgers after the `cycleExpiryDate`.
+     * @dev Reverts if the calling contract is not one of the project's terminals. This is the final call in the
+     * call chain described in `redeemParams()`.
+     * @param _data Standard Juicebox project redemption data.
+     * See https://docs.juicebox.money/dev/api/data-structures/jbdidredeemdata3_1_1/.
+     */
+    function didRedeem(JBDidRedeemData3_1_1 calldata _data) external payable terminalCheck redeemCycleCheck {
+        if (_data.projectId != campaign.projectId) revert IncorrectProjectID();
+
+        address payable redeemer = payable(_data.holder);
+        if (unwrap(pledgers.pledgerWeight[redeemer]) == 0) revert MustBePledger();
+        if (pledgers.hasBeenRefunded[redeemer]) revert AlreadyWithdrawnRefund();
+
+        // Calculates the refund bonus, using PRBMath, based on the time from cycle start and the size of the pledge
+        uint256 pledgerRefundBonus = unwrap(
+            mul(div(pledgers.pledgerWeight[redeemer], pledgers.totalPledgeWeight), wrap(campaign.totalRefundBonus))
+        );
+
+        // Sanity Check
+        if (address(this).balance < pledgerRefundBonus) revert InsufficientFunds();
+
+        // This redeeem function is now locked for the current redeeming address.
+        pledgers.hasBeenRefunded[redeemer] = true;
+
+        // Sending early pledger refund bonus.
+        (bool sendSuccess,) = redeemer.call{value: pledgerRefundBonus}("");
+        require(sendSuccess, "Failed to send refund bonus.");
+
+        emit CycleRefundBonusWithdrawal(redeemer, pledgerRefundBonus);
+    }
+
+    /**
+     * @notice This function is callable after cycleExpiryDate if goal is met. However, for
+     * contingency, where a redeemer is unable to retrieve funds, after a locked period, the
+     * Campaign Manager can withdraw funds for disbursement to remaining redeemers . The unlock
+     * time is hardcoded for two weeks after the cycleExpiryDate to give pledgers confidence
+     * that funds are programmatically guaranteed to remain in the contract for that time.
+     * @param receivingAddress Campaign Manager must call this function but can withdraw funds to
+     * another address if desired.
+     */
+    function creatorWithdraw(address payable receivingAddress, uint256 amount)
+        external
+        payable
+        onlyRole(CAMPAIGN_MANAGER_ROLE)
+    {
+        uint256 unlockTime = campaign.cycleExpiryDate + 1209600; // Two weeks after expiry date
+        require(
+            (hasCycleExpired() && isTargetMet()) || ((block.timestamp > unlockTime) && !isTargetMet()),
+            "Cycle must be expired and successful, or it must be passed the lock period."
```

Comment:
typo: `passed` should be `past`. Any reason not to use a custom error here?

Reply:
Dangit. Thanks. Done.

Reply:
I don’t know if there is a magical number to where custom errors get too long, but something like CycleIsActiveHasFailedOrFunctionIsLocked() felt too long, or you would need commas or other syntax to make it readable. If you think of a short one that conveys all that info, please send on over!

Reply:
What about this: `error CycleMustBeFundedAndClosedOrWaitForUnlock()`? Is that too long?

Reply:
I think your reasoning is sound. I was just curious of your reasoning between using them and not. A wrote a lot of code without custom errors, and I haven't found the right balance. I would keep what you have.